src/frontend/deployment_details.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/sirupsen/logrus"
+)
+
+var areDeploymentDetailsLoaded = false
+var deploymentDetailsMap = make(map[string]string)
+var log *logrus.Logger
+
+func init() {
+	initializeLogger()
+	go loadDeploymentDetails()
+}
+
+func initializeLogger() {
+	log = logrus.New()
+	log.Level = logrus.DebugLevel
+	log.Formatter = &logrus.JSONFormatter{
+		FieldMap: logrus.FieldMap{
+			logrus.FieldKeyTime:  "timestamp",
+			logrus.FieldKeyLevel: "severity",
+			logrus.FieldKeyMsg:   "message",
+		},
+		TimestampFormat: time.RFC3339Nano,
+	}
+	log.Out = os.Stdout
+}
```

Comment:
I copied this log-related chunk from https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/shippingservice/main.go#L46.